core/src/main/java/org/apache/calcite/sql/SqlUpdate.java

```diff
@@ -198,13 +201,8 @@ public SqlIdentifier getSourceAlias() {
       writer.keyword("AS");
       alias.unparse(writer, opLeft, opRight);
     }
-    if (sourceTable != null) {
-      writer.keyword("FROM");
```

Comment:
Another option to using "if (dialect == BIGQUERY)" statements throughout this function would be to move all of the SqlUpdate unparsing logic into the dialect-specific files. This would mean relocating the existing implementation into the SqlDialect class. And then overriding it with the BigQuery-specific logic in the BigQuerySqlDialect class.

Here's an example of where this is being done:
https://github.com/googleinterns/calcite/blob/master/core/src/main/java/org/apache/calcite/sql/SqlIntervalLiteral.java#L75

Reply:
I like the sound of this approach, I'll give it a try.